.github/workflows/docker-adhoc.yml

```diff
@@ -0,0 +1,47 @@
+name: Docker Manual Build & Push
+
+on:
+  workflow_dispatch:
+    inputs:
+      ref:
```

Comment:
<img width="1440" height="646" alt="Image" src="https://github.com/user-attachments/assets/72d8528b-64a7-4462-9441-1a399788d602" />

We may want to use the branch that we already have to pass into the dispatch.